### DISCUSSION TITLE
Why does frida_mode osx-lib forkserver timeout/not respond? How does forkserver work within frida mode?

### DISCUSSION THREAD

**Discussion by KraneBrain** [AFLplusplus/AFLplusplus#1679] [Q&A]
When running "gmake frida_persistent" within frida_mode/test/osx-lib the forkserver seems to fail. I verified that the entrypoint is hit. Inside src/instrument/instrument.c this comment is mentioned. "it is not until then that we start the fork-server and thus start executing in the child". How does frida start the forkserver? It appears the parent is waiting for data until timeout is reached. Any ideas?

     * Lastly, we don't enable pre-fetching back to the parent until we reach
     * our AFL_ENTRYPOINT, since it is not until then that we start the
     * fork-server and thus start executing in the child.

**Comment by vanhauser-thc** [comment]
MacOS comes with serious changes every update and most of the time breaking frida and other tools.
MacOS is not that well supported on our side as no one in the team is using it. A few things are known not to work and need someone to fix it (#1594).
For your issue the most likely fix is to update the frida version in GNUmakefile to the most current one and hope this fixes it.
@WorksButNotTested
